compiler/rustc_passes/src/naked_functions.rs

```diff
@@ -201,8 +209,18 @@ impl<'tcx> CheckInlineAssembly<'tcx> {
             }
 
             ExprKind::InlineAsm(asm) => {
-                self.items.push((ItemKind::Asm, span));
-                self.check_inline_asm(asm, span);
+                match asm.asm_macro {
+                    rustc_ast::AsmMacro::Asm => {
+                        self.items.push((ItemKind::InlineAsm, span));
+                    }
+                    rustc_ast::AsmMacro::NakedAsm => {
+                        self.items.push((ItemKind::NakedAsm, span));
+                        self.check_inline_asm(asm, span);
```

Comment:
this check is actually not needed any more: the parser will error out in all cases (invalid operands and invalid options). Should we keep it, or just remove it?

Reply:
Yes this should just be removed.